src/Orleans.Transactions/DistributedTM/TransactionAgent.cs

```diff
@@ -5,11 +5,52 @@
 using Orleans.Runtime;
 using Orleans.Concurrency;
 using System.Linq;
+using System.Threading;
+using System.Transactions;
 using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Orleans.Configuration;
 using Orleans.Transactions.Abstractions;
 
 namespace Orleans.Transactions
 {
+    public class TransactionAgentStatistics
+    {
+        public const string TransactionStartedPerSecondMetric = "TransactionAgent.TransactionStartedPerSecond";
+        //TransactionStartedPerSecond for last reporting period
+        public double TransactionStartedPerSecond { get; private set; }
+
+        public long TransactionStartedCounter { get; set; }
+
+        public DateTime LastReportTime { get; private set; } = DateTime.UtcNow;
+        private readonly ITelemetryProducer telemetryProducer;
+        private readonly PeriodicAction monitor;
+        public TransactionAgentStatistics(ITelemetryProducer telemetryProducer, IOptions<StatisticsOptions> options)
+        {
+            this.telemetryProducer = telemetryProducer;
+            this.monitor = new PeriodicAction(options.Value.PerfCountersWriteInterval, this.ReportMetrics);
+        }
+
+        public void TryReportMetrics()
+        {
+            this.monitor.TryAction(DateTime.UtcNow);
+        }
+
+        private void ReportMetrics()
+        {
+            var now = DateTime.UtcNow;
+            var txStarted = TransactionStartedCounter;
+            var timelapseSinceLastReport = now - this.LastReportTime;
+            if (timelapseSinceLastReport.TotalSeconds <= 1)
```

Comment:
Suggest using (txStarted * 1000) / by time delta in ticks rather than seconds, in case sub-second report windows are configured.  It's also more accurate.